### PR TITLE
Added feature for chaining manifests together (fixes #19). Also fixed…

### LINKED ISSUE #19: Allow for chaining of sofe manifests.
For example:
```
//config.js
System.config({
  sofe: {
    manifestUrl: "/sofe-manifest.json"
  }
}
```

```
// sofe-manifest.json
{
  "sofe": {
    "manifestUrl": "https://cdn.canopytax.com/sofe-manifest.json"
  }
}
```

```
// https://cdn.canopytax.com/sofe-manifest.json
{
  "sofe": {
    "manifest": {
      "serv1": "...."
    }
  }
}
```

### PR DESCRIPTION
**Pull request by joeldenning**
… #20.

### CODE REVIEW

**Comment by joeldenning** [comment]
@blittle the code cov comment is out of date because I force pushed, but the comment wasn't updated / reposted. You can see the true code coverage at https://codecov.io/github/CanopyTax/sofe?branch=issues-19-and-20

File: src/manifest.js [hunk 1]
```diff
@@ -35,21 +35,64 @@ export function getManifest(config) {
...
-						};
 
-						if (hasWindow) {
-							window.sofe.cachedRemoteManifest = cachedRemoteManifest;
+					visitedManifestUrls.push(config.manifestUrl);
+
+					if (json && json.sofe) {
+						// First check if this manifest points to another manifest
+						if (typeof json.sofe.manifestUrl === 'string') {
+							if (visitedManifestUrls.indexOf(json.sofe.manifestUrl) >= 0) {
```
Comment (blittle) [inline src/manifest.js]: :+1: on the circular check

File: src/manifest.js [hunk 2]
```diff
@@ -35,21 +35,64 @@ export function getManifest(config) {
...
+								reject(
+									new Error(`Cannot load manifest -- circular chain of sofe manifests, '${json.sofe.manifestUrl}' detected twice in chain.`)
+								)
+							} else {
+								getManifest(json.sofe, visitedManifestUrls)
+								.then(chainedManifest => {
+									if (config.manifest === Object(config.manifest)) {
```
Comment (blittle) [inline src/manifest.js]: What is this check for?

Comment (blittle) [inline src/manifest.js]: Nm, see the note below

**Comment by blittle** [comment]
:+1:

**Pull request merged**